pyro/ops/stats.py

```diff
@@ -261,6 +261,61 @@ def quantile(input, probs, dim=0):
     return quantiles if probs.shape != torch.Size([]) else quantiles.squeeze(dim)
 
 
+def weighed_quantile(input, probs, log_weights, dim=0):
```

Comment:
Would you mind adding type hints for new code?
```py
from typing import List

def weighted_quantile(
    input: torch.Tensor,
    probs: List[float],
    log_weights: torch.Tensor,
    dim: int = 0,
) -> torch.Tensor:
    ...
```

Reply:
Thank you for the review @fritzo! I've added testing and type hints.